include/union_room.h

```diff
@@ -85,56 +85,56 @@ struct UnkStruct_Group
     u8 field_15;
 };
 
```

Comment:
(comment on line 31) is there any clue of sizeof(struct UnkStruct_Main0)? 
I tend to believe that field_0 in `struct UnkStruct_URoom`, `struct UnkStruct_Group`, `struct UnkStruct_Leader`are pointing to different structs, or just one struct holding max size of all those params passed to `AllocZeroed`. e.g. 0x200 indicates `struct UnkStruct_x20 arr[16];`

And, can you check `struct UnkStruct_Main4` and `struct UnkStruct_Main8` as well?

Reply:
does the current setup make sense? maybe we can make them something like `struct XXX { struct UnkStruct_xxx arr[0] };`?